scripts/build_osm_network.py

```diff
@@ -109,74 +110,69 @@ def set_substations_ids(buses, distance_crs, tol=2000):
                 buses.loc[buses.index[close_nodes], "station_id"] = sub_id
 
 
-def set_lines_ids(lines, buses, distance_crs):
+def set_lines_ids(lines, buses):
     """
     Function to set line buses ids to the closest bus in the list.
     """
     # set tqdm options for set lines ids
-    tqdm_kwargs_line_ids = dict(
-        ascii=False,
-        unit=" lines",
-        total=lines.shape[0],
-        desc="Set line bus ids ",
-    )
 
     # initialization
     lines["bus0"] = -1
     lines["bus1"] = -1
 
-    busesepsg = buses.to_crs(distance_crs)
-    linesepsg = lines.to_crs(distance_crs)
-
-    for i, row in tqdm(linesepsg.iterrows(), **tqdm_kwargs_line_ids):
-        # select buses having the voltage level of the current line
-        buses_sel = busesepsg[
-            (buses["voltage"] == row["voltage"]) & (buses["dc"] == row["dc"])
-        ]
+    for key, lines_sel in lines.groupby(["voltage", "dc"]):
+        buses_sel = buses.query(f"voltage == {key[0]} and dc == {key[1]}")
```

Comment:
just to understand this a bit better. why do you need to group the lines by "voltage", "dc"?

something like this should do the trick
```python
# Merge the `lines` and `buses` DataFrames on the common columns ("voltage" and "dc")
merged_data = lines.merge(buses, on=["voltage", "dc"])
```

Reply:
because the lines corresponding to a voltage level and dc/ac status shall be matched with the corresponding types of buses.
If you have smarter way, happy to hear alternatives :)

Reply:
Nice, good to know that we have that information in the buses. Alternative would be just the snippet I shared, does the groupby and query in one.